samples/widgets/sales-dashboard-widgets/RevenueByProductWithUserPreferences/src/RevenueByProductWithUserPreferences.jsx

```diff
@@ -0,0 +1,138 @@
+            ['ALL', 'ALL', 'Europe', 'Analytics', 3267364.63, 292, 181],
+            ['ALL', 'ALL', 'Europe', 'APIManager', 3249096.11, 310, 172],
+            ['ALL', 'ALL', 'Europe', 'ESB', 2078261.1, 175, 114],
+            ['ALL', 'ALL', 'Europe', 'IdentityServer', 993996.02, 84, 43],
+            ['ALL', 'ALL', 'Europe', 'IOTServer', 0, 0, 0],
+            ['ALL', 'ALL', 'LATAM', 'Analytics', 370146.06, 39, 23],
+            ['ALL', 'ALL', 'LATAM', 'APIManager', 290146.06, 32, 16],
+            ['ALL', 'ALL', 'LATAM', 'ESB', 1020146.06, 33, 15],
+            ['ALL', 'ALL', 'LATAM', 'IdentityServer', 0, 0, 0],
+            ['ALL', 'ALL', 'LATAM', 'IOTServer', 0, 0, 0],
+            ['ALL', 'ALL', 'Middle East', 'Analytics', 130859.48, 13, 9],
+            ['ALL', 'ALL', 'Middle East', 'APIManager', 367368.36, 31, 15],
+            ['ALL', 'ALL', 'Middle East', 'ESB', 436754.93, 51, 26],
+            ['ALL', 'ALL', 'Middle East', 'IdentityServer', 928227.84, 95, 64],
+            ['ALL', 'ALL', 'Middle East', 'IOTServer', 91472.91, 14, 8],
+            ['ALL', 'ALL', 'North America', 'Analytics', 324886.4, 37, 23],
+            ['ALL', 'ALL', 'North America', 'APIManager', 774886.4, 70, 44],
+            ['ALL', 'ALL', 'North America', 'ESB', 1134886.4, 40, 24],
+            ['ALL', 'ALL', 'North America', 'IdentityServer', 390128.28, 30, 12],
+            ['ALL', 'ALL', 'North America', 'IOTServer', 0, 0, 0],
+            ['ALL', 'ALL', 'ROW', 'Analytics', 114522.14, 13, 8],
+            ['ALL', 'ALL', 'ROW', 'APIManager', 214522.14, 19, 10],
+            ['ALL', 'ALL', 'ROW', 'ESB', 174522.14, 15, 7],
+            ['ALL', 'ALL', 'ROW', 'IdentityServer', 224522.14, 25, 17],
+            ['ALL', 'ALL', 'ROW', 'IOTServer', 0, 0, 0]
+        ];
+
+        this.state = {
+            data: this.overallProductData,
+            width: props.glContainer.width,
+            height: props.glContainer.height
+
+        };
+
+        this.metadata = {
+            names: ['Country', 'Country Code', 'Region', 'Product', 'Revenue', 'Downloads', 'Customers'],
+            types: ['ordinal', 'ordinal', 'ordinal', 'ordinal', 'linear', 'linear', 'linear']
+        };
+
+        this.barChartConfig = {
+            x: 'Product',
+            charts: [{type: 'bar', y: 'Revenue', fill: '#10c469'}],
+            maxLength: 5,
+            width: props.glContainer.width,
+            height: props.glContainer.height,
+            animate: true
+        };
+
+        this.setReceivedMsg = this.setReceivedMsg.bind(this);
+        this.handleResize = this.handleResize.bind(this);
+        this.props.glContainer.on('resize', this.handleResize);
+    }
+
+    handleResize() {
+        this.setState({width: this.props.glContainer.width, height: this.props.glContainer.height});
+    }
+
+    componentWillMount() {
+        super.subscribe(this.setReceivedMsg);
+    }
+
+    setReceivedMsg(receivedMsg) {
+        if (receivedMsg.selectedRegion === "ALL") {
```

Comment:
Please use single quotes for 'ALL'

Reply:
I think it is better to keep it as it is to be inline with the other widgets

Reply:
ESLint specifications enforce to use single quotes for strings. It's always better to follow those

Reply:
ESLint is just a JavaScript linting tool, it does not enforce any rules if not specified by us, single quotes vs double quotes in JSX is just a matter of preference, it is better to use single quotes here so the code is inline with the code base. See RevenueByProduct.jsx

Reply:
We have an [ESLint ruleset](https://github.com/wso2/carbon-dashboards/blob/master/components/dashboards-web-component/.eslintrc) in carbon-dashboards-web-component. If you would have enabled that, this would have been a warning

Reply:
The whole purpose of a Linter is to maintain the consistency of the code base please refer to [RevenueByProduct.jsx](https://github.com/wso2/carbon-dashboards/blob/0fc856d46e6534e33332b49acf7e4542c8dd1952/samples/widgets/sales-dashboard-widgets/RevenueByProduct/src/RevenueByProduct.jsx#L108)

Reply:
done in 4cc2bf516aa03beece86f9278ee88e1b22c97aba